oidc-appsupport/src/wasmJsMain/kotlin/org/publicvalue/multiplatform/oidc/appsupport/PlatformOidcCodeAuthFlow.wasmJs.kt

```diff
@@ -0,0 +1,73 @@
+package org.publicvalue.multiplatform.oidc.appsupport
+
+import io.ktor.http.*
+import kotlinx.browser.window
+import kotlinx.coroutines.delay
+import org.publicvalue.multiplatform.oidc.OpenIdConnectClient
+import org.publicvalue.multiplatform.oidc.OpenIdConnectException
+import org.publicvalue.multiplatform.oidc.flows.AuthCodeResponse
+import org.publicvalue.multiplatform.oidc.flows.AuthCodeResult
+import org.publicvalue.multiplatform.oidc.flows.CodeAuthFlow
+import org.publicvalue.multiplatform.oidc.types.AuthCodeRequest
+import org.w3c.dom.Window
+
+
+actual class PlatformCodeAuthFlow(
+    client: OpenIdConnectClient
+) : CodeAuthFlow(client) {
+
+
+    actual override suspend fun getAuthorizationCode(request: AuthCodeRequest): AuthCodeResponse {
+        //window.location.replace(request.url.toString())
+        val loginWindow = window.open(request.url.toString())
+
+        // WAIT FOR EXTERNAL INPUT
+        val resultParameters = waitForRedirectUri(loginWindow)
+        return AuthCodeResponse.success(
+            AuthCodeResult(
+                code = resultParameters!!["code"],
+                state = resultParameters["state"]
+            )
+        )
+
+    }
+
+    suspend fun waitForRedirectUri(loginWindow: Window?): Parameters? {
+        while (true) {
+            var parameters: Parameters? = null
+            jsCatch {
```

Comment:
Why do we need an additional jsCatch here?
Can't we just catch Throwable or dynamic as mentioned here https://discuss.kotlinlang.org/t/how-to-catch-non-kotlin-exception/2404/14

Reply:
We can only catch Throwable, I'm trying to catch dynamic but i get an error: "Unsupported [Dynamic types are not supported in this context". I think Throwable is enough